inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_binary_elementwise.cpp

```diff
@@ -2,87 +2,108 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include "dsr_tests_common.hpp"
+
 #include <functional_test_utils/layer_test_utils.hpp>
 #include <ngraph_functions/builders.hpp>
 #include <vpu/ngraph/operations/dynamic_shape_resolver.hpp>
 
 namespace {
 
-using DataType = ngraph::element::Type_t;
-using DataDims = ngraph::Shape;
+using namespace LayerTestsUtils::vpu;
+
+struct BinaryEltwiseShapes {
+    DataShapeWithUpperBound inputShape0;
+    DataShapeWithUpperBound inputShape1;
+};
 
-using Parameters = std::tuple<
+using BinaryElementwiseParameters = std::tuple<
     DataType,
-    DataDims,
-    DataDims,
+    BinaryEltwiseShapes,
     ngraph::NodeTypeInfo,
     LayerTestsUtils::TargetDevice
 >;
 
-class DSR_BinaryElementwise : public testing::WithParamInterface<Parameters>,
-        public LayerTestsUtils::LayerTestsCommon {
+class DSR_BinaryElementwiseBase : public testing::WithParamInterface<BinaryElementwiseParameters>,
+                                  public DSR_TestsCommon {
 protected:
-    void SetUp() override {
-        const auto& parameters = GetParam();
-        const auto& dataType = std::get<0>(parameters);
-        const auto& dataDims0 = std::get<1>(parameters);
-        const auto& dataDims1 = std::get<2>(parameters);
-        const auto& eltwiseType = std::get<3>(parameters);
-        targetDevice = std::get<4>(parameters);
-
-        const auto input0 = std::make_shared<ngraph::opset3::Parameter>(dataType, dataDims0);
-        const auto input1 = std::make_shared<ngraph::opset3::Parameter>(dataType, dataDims1);
+    ngraph::NodeTypeInfo m_eltwiseType{};
+    DataType m_inDataType = ngraph::element::dynamic;
+
+    InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo &info) const override {
+        // Avoid division by zero
+        const auto opForWhichAvoidingIsNecessary =
+                m_eltwiseType == ngraph::opset3::Divide::type_info ||
+                m_eltwiseType == ngraph::opset3::Power::type_info;
+        const auto isDataInput = m_shapes.find(info.name()) == m_shapes.end();
+        if (opForWhichAvoidingIsNecessary && isDataInput) {
+            return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 5, 1, 1);
+        }
+        return DSR_TestsCommon::GenerateInput(info);
```

Comment:
Is it easier just avoid zeros for any input in generation?

Reply:
Moved this to the common part.